### ISSUE TITLE
New theme using Battle Mode as main interface

### ISSUE THREAD

**Issue by dragonjet** [dragonjet/KC3Kai#187]
Since the battle mode almost included everything, make amends to the interface so it can be fully promoted to being the main interface, removing the need for battle mode detection.

This will go into a new theme. Default will remain as is, with normal/battle modes.

**Comment by LostLogia4** [comment]
May I get a screenshot of this "Battle Mode" interface?

**Comment by dragonjet** [comment]
example battle mode via PvP. same interface on sorties, just with node list on top-right

![ss 2015-07-09 at 08 08 56](https://cloud.githubusercontent.com/assets/557312/8585236/d84c6a3e-2611-11e5-88d0-bd4407d62415.png)

**Comment by McDohl** [comment]
from what i test, the enemy data as in detect, engage, contact and air battle not show if theres no enemy that can bring fighter / bomber. instead it will show blank panel. 

So my guess is, it hasn't implemented yet on current version. do your best KC3Kai team :+1:

**Comment by dragonjet** [comment]
oh you're right I never noticed that bug xD

tested 1-1 with no air-capable ships and it really was blank :open_mouth:

thanks for the support :smile:

**Comment by McDohl** [comment]
here is my report for latest version (till screenshot month index fixed): 
![battle mode](https://cloud.githubusercontent.com/assets/13024936/8585888/62af69d8-2612-11e5-8bce-bab622e31c42.JPG)

* From that pic its shown that if your fleet has no planes but still get AS+ which is weird. meanwhile from wiki, If both sides launch no planes, then the rest of the battle is treated the same as if Air Parity was achieved.

![contact](https://cloud.githubusercontent.com/assets/13024936/8585973/ab6fc112-2613-11e5-8fb5-29ee0262121c.png)
* from the game i get contact, meanwhile Battle mode reported it as No contact

**Comment by dragonjet** [comment]
1) **AS+** hm yes I noticed this as well ever since the beginning of implementation of air battles, which can also be seen on current v9 WebStore versions in battle history. I never bothered to change it since I thought that really was the case, but if the wiki says so I might need to revisit the articles

2) I see a possible cause to this issue, but I need to find a way to reproduce this environment. I will just copy your CVL setup and sortie to 1-1 and hopefully I also get a contact. Background story for this is that contact happened on the second air phase. Currently we are just checking air phase 1, which I thought is the only thing we needed to check.

**Comment by bamboo3250** [comment]
According to Wiki, the condition for AS+ is FP ≥ 3*EFP, which is satisfied if fighter power of both side are 0. So here the wiki contradict to itself. Is there any discussion about this on wiki before? If not, we should create a thread for that.

**Comment by dragonjet** [comment]
oh yea, 0=0, which is also what the game data returns. I am not counter-checking with fellow viewers on what they do in this case

**Comment by McDohl** [comment]
1. I see, since its bugging me to no end might try sparkling CA & BB sooner or later before 4-5 run and see if i get double attack on day battle

2. might test it later on 1-6 with LHA and CAV and hoping if i get contact within few runs

**Comment by dragonjet** [comment]
I checked [ElectronicObserver's Air Superiority values](https://github.com/andanteyk/ElectronicObserver/blob/master/ElectronicObserver/Data/Constants.cs#L350). they also use the same values.

**1** is *Air Supremacy* which is unfortunately what the game returns even when there's no air battle

**Comment by LostLogia4** [comment]
Well, the air battle phase must be initiated to claim the air supremacy to begin with. When both sides has no planes at all, you're stuck with air parity.

**Comment by dragonjet** [comment]
well maybe, since we can show the all the plane counts for both sides, if fighter count for both side is zero then force the air battle text as "None", would that work? or maybe there are other mechanics at work?

**Comment by LostLogia4** [comment]
Off topic, but what's those number beside the guns for?
Speaking of plane slots, they're only applicable for... well, planes.

**Comment by bamboo3250** [comment]
@LostLogia4 I see... that makes sense to me. 
@dragonjet I think the condition should be fighter count for both side is 0 AND `stage2 == null` then there is no air battle.

**Comment by McDohl** [comment]
i dont know if i should create a new issue or not but since my previous comment is in this issue then might as well post it here.

![4-5](https://cloud.githubusercontent.com/assets/13024936/8608681/759e8636-26c7-11e5-9413-dc8f208b60e9.png)

I'm using the latest build version, but seems like contact still not working properly

**Comment by dragonjet** [comment]
maybe we should track it on a new issue. I was almost sure that would have fixed it, as we are also using the same way as [ElectronicObserver's Contact](https://github.com/andanteyk/ElectronicObserver/blob/master/ElectronicObserver/Data/Battle/Phase/PhaseAirBattle.cs#L146)

**Comment by dragonjet** [comment]
I derped. pushed the wrong button

**Comment by LostLogia4** [comment]
The other bug aside, may I get a preview of this enhanced battle mode?

**Comment by bamboo3250** [comment]
We still have to find solutions for some tough issues before we can do this one. For example #312. You can join the discussion to think of the best approach for it.

**Comment by LostLogia4** [comment]
Ah, that... nameship spillover... I'll take a look... not sure I'll be of much help, so I ignored that bug until now.

**Comment by ethrundr** [comment]
the ship listing on Natsuiro still not work?
since I'm trying it now and doesn't see it appearing. just want to confirm whether is error on my side or not implemented yet

**Comment by dragonjet** [comment]
working on it right now XD

![ss 2015-08-08 at 03 52 47](https://cloud.githubusercontent.com/assets/557312/9149634/8caecd2e-3de5-11e5-8991-e37cbd787d2b.jpg)

also, not possible to quick switch between themes. both use different assets and scripts

**Comment by bamboo3250** [comment]
Wow, that looks really cool @ A @

**Comment by Crazyteitoku** [comment]
I can totally imagine someone liking this sort of theme @dragonjet - considering how long lived this feature request was. :laughing: :+1:

**Comment by dragonjet** [comment]
nico nico nii~

the good this abt natsuiro is that it's compatible with vertical using the same code, just different css positions, but same boxes XD

![ss 2015-08-08 at 04 57 19](https://cloud.githubusercontent.com/assets/557312/9149814/9a3a3b64-3dee-11e5-8bce-edb8ddd6f57e.jpg)

**Comment by McDohl** [comment]
question for vertical, is it 300px or 420px or have both version.

**Comment by bamboo3250** [comment]
@McDohl from the screenshot, it looks like 320px. Still need confirmation from @dragonjet though :stuck_out_tongue:

**Comment by ethrundr** [comment]
wow now that's look great....
well hope it committed soon :laughing:

**Comment by dragonjet** [comment]
whats the diff between 300 vs 420 though? xD just the height?

**Comment by bamboo3250** [comment]
@dragonjet the position of boxes is rearranged quite a lot xD

**Comment by dragonjet** [comment]
I'm not sure whats happening XD natsuiro needs 420px? :open_mouth:
iirc its just because the text is small thats why we had 420px
i think natsuiro vertical can suffice without it?
or there's an underlying cause I do not know for the `420px`

**Comment by bamboo3250** [comment]
420px will perfectly fit with large screen 1920x1080. That is the only reason I chose 420px for large version panel. However, Natsuiro might not need to be as big as 420px. From what I see in the screenshot, the text is clear enough to me :+1:

**Comment by McDohl** [comment]
will see first, hope it not need 420px. since 300px (default theme) for 1080p user (well maybe it just me xD) is like too small :smile:

**Comment by nicky008** [comment]
Mine is 1366x768, stick to the horizontal mode.

/me is kicked outside the discussion. :dizzy_face:

But...but I think I also have no problem for the 420px on my 2nd wide screen though...

**Comment by ethrundr** [comment]
my resolution is 1440 x something
so, horizontal for me :smile:

**Comment by McDohl** [comment]
after some testing for https://github.com/dragonjet/KC3Kai/commit/6b6bb9cb6f25abaa2f80b5645d695b1936f8ca02 , 
* isnt it supposed to show after we collect the information from api_start2 ?
![outofplace](https://cloud.githubusercontent.com/assets/13024936/9152917/dd80d0fa-3e6f-11e5-8564-55a4f1c042bb.png)

heres a dark/ natsuiro theme preview on 1920x1080

![bwkiller](https://cloud.githubusercontent.com/assets/13024936/9153004/7c9ed0a8-3e73-11e5-8b4d-e13fffec2a0d.jpg)

and my suggestion for vertical mode, how bout put that crafting / expedition helper below quest list?

**Comment by dragonjet** [comment]
@McDohl ah there are some parts still on designing phase, so I need to keep the full interface shown even without data, have to refresh often to check how the layout progress and don't wanna wait for API calls at all times XD

the crafting box should actually be on the activity tab. when you do craft something, it will automatically switch to the "Activity". That's the tabbing on the left with Basic, Battle, Activity

the expedition helper, I'm not sure yet xD
we'll need to decide which features go to the panel and which ones go to strategy room

**Comment by McDohl** [comment]
@dragonjet ah I see, 

after looking more at natsuiro, here is my suggestion:
* can you make Player comment on natsuiro/dark bold? it's kinda hard to see with that thin lines xD.
* HQ level exp maybe put Total, Next and Current like @ReiFan49 did to default theme. Or maybe you haven't implemented it yet xD
* screenshot button is like lonely there at bottom, instead why not put it beside butai icon?

**Comment by nicky008** [comment]
@McDohl Thx for your pull'n me back... :)

IMO, both 300px and 420px isn't so very difference in the aspect of font size and (icon) picture size. Both have shown same (font and picture) size, which might made some people felt uncomfortable.

**Comment by dragonjet** [comment]
hopefully they just use background if they don't want extra solid color spaces xD

**Comment by McDohl** [comment]
@dragonjet sadly i'm not a fond of using background for panels xD

**Comment by dragonjet** [comment]
this doesn't sound good. been hearing thunder and we got some rains goin on here. just in case power goes out later I cant upload v15 to webstore.

hopefully there won;t be any black out. even tmr, can't affort this

**Comment by nicky008** [comment]
@dragonjet 2.33 hours left, may the RNG force be with you. :+1:

**Comment by dragonjet** [comment]
minus the time I actually have to upload it around 30 mins early, and the time I have to write changelogs and whatsoevers :open_mouth:

**Comment by bamboo3250** [comment]
Hopefully @dragonjet still daijoubu :sob:

**Comment by ethrundr** [comment]
anyway, is the screenshot during DMM direct gameplay still cannot be used?
also, @dragonjet, have decided on the new hotkey for screenshot instead of F9?

**Comment by dragonjet** [comment]
yep, DMM script injection not yet done :disappointed: 

wait, were we changing the screenshot hotkey? xD

**Comment by McDohl** [comment]
I heard OSX can't capture using F9 key

**Comment by dragonjet** [comment]
hmm what's a safe button to use?

**Comment by McDohl** [comment]
i only heard it from @ethrundr , maybe ask his further assistance since i not own a mac
